### ISSUE TITLE
Ionic Components - Desktop support

### ISSUE THREAD

**Issue by Newbie012** [ionic-team/ionic-framework#15500]
# Feature Request

**Describe the Feature Request**
**TL;DR** - Components for Desktop.
Over the past year, Ionic has outgrown and now offers more than just building mobile apps.
Since the release of Ionic 4 beta and stencil, It's much more convenient to build PWA using Ionic.
Yet, there's one thing that still prevents me to move to Ionic and that is Desktop support. Ionic has lots of components, but there are some of them that aren't exactly fit desktop mode or just doesn't exists. Currently, I'm using Angular and Angular Material (which offers desktop support).

**Describe Preferred Solution**
- *Menu/Select Input* - Many times we need some sort of menu. It exists on [Material UI](https://material-components.github.io/material-components-web-catalog/#/component/menu) and so on [Angular Material Select](https://material.angular.io/components/select/overview). It could have been nice if there was something like:
  ```
  <ion-select interface="dropdown">
    <ion-select-option>...</ion-select-option>
  </ion-select >
  ```
- *DateTime* - The current available DateTime will pop up from the bottom of the screen. This is the desired behavior on mobile, but is it on a desktop? IMHO [Material Angular DatePicker](https://material.angular.io/components/datepicker/overview) offers a convenient way on a desktop to display the Calendar.
- *Text Fields* - I'm not sure on how it should be implemented on iOS, but Material has a [Text Field](https://material-components.github.io/material-components-web-catalog/#/component/text-field) which suggests in multiple ways on how to display the text field (e.g filled, outlined).

**Describe Alternatives**
Currently, my alternative is to use both Angular and Angular Material, which is a bummer.

**Comment by patrickmcd** [comment]
A date picker in the form of a calendar would be great. The [Touch UI mode version](https://material.angular.io/components/datepicker/overview#touch-ui-mode), which opens as a bigger modal, is what I'd expect.

A simple idea for making pickers more usable on desktop would be to make each picker column react to the scroll wheel. Manipulating the dials is a pita by dragging with a mouse.

Helper / error text would be nice for inputs, and so would input icons.

I don't see a boilerplate difference between your suggestion and `ion-select` with the popover interface.

**Comment by Newbie012** [comment]
@patrickmcd About the boilerplate difference, you're right. The reason I wrote this was because the last time I used Ionic, I didn't have the option to use `interface="popover"`, and I had to create a new component just to display it. After I reviewed the docs again, I saw that there's already an option to display the `ion-select` as a popover. Yet, it's still not the desired behavior.

**Comment by RHinderiks** [comment]
Agree with this.

**Comment by jcn-ithaca** [comment]
Another feature that would be nice for desktop is support for dragging a modal around by its toolbar, so you can see the stuff behind it (e.g. it is showing you help content about the current page). We have a fairly complex Ionic 4 desktop app, and the fixed position of modals is a bit painful.

**Comment by etinif** [comment]
How is this issue still unresolved? Does Ionic care for desktop support? It's totally unusable to build admin dashboards for bigger screens. Forms are awful looking overall, selects appear at the bottom, etc. Ionic probably has the ugliest UI kit on the web/desktop lol.

**Comment by bchehraz** [comment]
Any idea when v6 would be ready to ship? The date and time pickers for v6 look amazing.

Been looking for alternatives for a TimePicker component if anyone has any suggestions, the TimePicker in v5 is very kludgy for desktop.
